internal/providers/github/properties/fetcher.go

```diff
@@ -27,14 +26,20 @@ import (
 )
 
 // GhPropertyWrapper is a function that fetches a property from the GitHub API
-type GhPropertyWrapper func(ctx context.Context, ghCli *go_github.Client, name string) (map[string]any, error)
+type GhPropertyWrapper func(
+	ctx context.Context, ghCli *go_github.Client, lookupProperties *properties.Properties,
+) (map[string]any, error)
 
 // GhPropertyFetcher is an interface for fetching properties from the GitHub API
 type GhPropertyFetcher interface {
+	ghPropertyFetcherBase
```

Comment:
I guess we wouldn't have to split the interface and could just make the per-entity structures inherit the base structure to satisfy the GhPropertyFetcher interface..?